### PR TITLE
Init CI, nix shells and an MLIR derivation

### PR DESCRIPTION
**Pull request by doug-q**
While reviewing, please do review the repository settings. I have intended to copy them from hugr, except that I am allowing PRs to rebase.

See the develop branch for where this is going.

@croyzor may I ask you to review?

### CODE REVIEW

File: nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch
Comment (croyzor) [inline nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch:26]: Why is this written as a generator? We could move the `else` branch below into a catch all case and avoid the for loop

File: nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch
Comment (croyzor) [inline nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch:51]: The same function is being added twice to the same file?

**Comment by croyzor** [comment]
As for the repository settings, I don't have permission to view them or add myself as a reviewer, so something is missing!

File: nix/mlir/default.nix
Comment (ss2165) [inline nix/mlir/default.nix:68]: Could you add the sources of each patch and a comment saying what each one does

**Review by ss2165 — COMMENTED** [review]
👍

**Comment by ss2165** [comment]
Repository settings seem fine

File: nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch
Comment (doug-q) [reply nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch:26]: I don't disagree but this is how it came from upstream (nixpkgs/pkgs/development/compilers/llvm/git/llvm/llvm-lit-cfg-add-libs-to-dylib-path.patch). I think it's best to leave it as is.

File: nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch
Comment (doug-q) [reply nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch:51]: Agree, will remove the second.

File: nix/mlir/default.nix
Comment (doug-q) [reply nix/mlir/default.nix:68]: Will do, I will copy the comments from nixpkgs.

File: .gitignore
Comment (doug-q) [reply .gitignore:1]: I don't think it's redundant: I like to use "/_build-blah" for build directories and this is the only pattern that matches that.

Comment (croyzor) [reply .gitignore:1]: This is extreme nitpicking, but what's wrong with `/_build*/`?

Comment (doug-q) [reply .gitignore:1]: Awkward, I somehow had thought the square brackets meant optional here. Thanks :)

**Comment by doug-q** [comment]
@croyzor Can you look again?

**Review by croyzor — APPROVED** [review]

**Comment by croyzor** [comment]
> At least 1 approving review is required by reviewers with write access.

As discussed, we should default to giving write access to the `tket` team for these repos. Once that's done, you should be able to merge! 🙂

**Pull request merged**
